sal-service/src/main/java/org/ow2/proactive/sal/service/repository/LocationRepository.java

```diff
@@ -40,4 +41,8 @@ public interface LocationRepository extends JpaRepository<Location, String> {
     @Transactional(readOnly = true)
     @Query(value = "SELECT id FROM Location WHERE id NOT IN (SELECT location.id FROM NodeCandidate GROUP BY location.id)")
     List<String> getOrphanLocationIds();
+
+    @Modifying(clearAutomatically = true)
+    @Query(value = "DELETE FROM Location WHERE id NOT IN (SELECT location.id FROM NodeCandidate GROUP BY location.id)")
```

Comment:
orphanRemoval is possible with @OneToMany decorators and not @ManyToOne, since the location class does not define NodeCandidate parameters, hence no OneToMany relation and we can't use the automatic OrphanRemoval defined by persistence. 

I guess that's why you created a separate method to do that manually by creating a query to collect them and then removing them one by one, I just wanted to improve on this and made it one single query that is faster ;)

Reply:
You're right. Fair enough then. :)